pkg/initializer/validator/datavalidator.go

```diff
@@ -68,6 +68,10 @@ func (d *DataValidator) backendPath() string { return filepath.Join(d.snapDir(),
 func (d *DataValidator) Validate(mode Mode, failBelowRevision int64) (DataDirStatus, error) {
 	status, err := d.sanityCheck(failBelowRevision)
 	if status != DataDirectoryValid {
+		// TODO: To be removed when backup-restore supports restoration of single member in multi-node etcd cluster.
+		if d.OriginalClusterSize > 1 && !miscellaneous.IsBackupBucketEmpty(d.Config.SnapstoreConfig, d.Logger) {
```

Comment:
I suggest to add this check to the `sanityCheck` function as this one contains the main logic.

Reply:
I tried but it is getting little complicated, so let's not invest time in it as we are anyway going to remove this logic in upcoming releases. Hope that's fine.